### PR TITLE
Authorize consumers of generic events for outbound

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

Added authorization of consumers for generic cloud events

## Related Issue(s)
- #305

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
~~- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)~~ automated tests for events handled in another issue
- [x] All tests run green

### CODE REVIEW

File: src/Events/Extensions/CloudEventExtensions.cs [hunk 1]
```diff
@@ -19,5 +20,26 @@ public static string Serialize(this CloudEvent cloudEvent, CloudEventFormatter f
...
+        /// Retrieves the resource instance from the cloud event. Returns null if it isn't defined.
+        /// </summary>
+        public static string GetResourceInstance(this CloudEvent cloudEvent) => cloudEvent["resourceinstance"]?.ToString();
+
+        /// <summary>
+        /// Sets the resource extension attribute of the cloud event
+        /// </summary>
+        public static void SetResource(this CloudEvent cloudEvent, string resource)
+        {
+            if (!string.IsNullOrEmpty(resource))
```
Comment (SandGrainOne) [inline src/Events/Extensions/CloudEventExtensions.cs:39]: This if check should ideally be reflected in the method name. `SetResourceIfNull` or something.

Comment (acn-sbuad) [reply src/Events/Extensions/CloudEventExtensions.cs:39]: We don't actually check that resource isn't defined on the cloud event, only that the input isn't empty. That being said, we probably should. Renaming and ensuring resource hasn't been defined before setting the value

File: src/Events/Services/OutboundService.cs [hunk 2]
```diff
@@ -138,6 +146,16 @@ private static CloudEventEnvelope MapToEnvelope(CloudEvent cloudEvent, Subscript
             return cloudEventEnvelope;
         }
 
+        private static string GetAuthorizationCacheKey(string resource, string consumer)
+        {
+            if (resource == null)
```
Comment (SandGrainOne) [inline src/Events/Services/OutboundService.cs:151]: Is there a chance for resource to be null? Would it even work with a cache key of null?

File: src/Events/Extensions/CloudEventExtensions.cs (hunk 1 above)
Comment (SandGrainOne) [reply src/Events/Extensions/CloudEventExtensions.cs]: oh, sorry, my comment was a mess.

My primary concern was the hidden if. Set methods should work like a setter and set the value. Think of how you would have handled this if Resource was a regular property.

Comment (acn-sbuad) [reply src/Events/Extensions/CloudEventExtensions.cs]: Removed the hidden if check, and the method name reflects what it does. We don't override the resource name if it is already defined. Do you think we need an argument null exception here in addition to stop from setting an empty resource value?

File: src/Events/Services/OutboundService.cs (hunk 2 above)
Comment (acn-sbuad) [reply src/Events/Services/OutboundService.cs:151]: Maybe we could rename to setResourceIfNotDefined to be more precise?

File: src/Events/Extensions/CloudEventExtensions.cs (hunk 1 above)
Comment (acn-sbuad) [reply src/Events/Extensions/CloudEventExtensions.cs]: Maybe we could rename to setResourceIfNotDefined to be more precise?

Comment (SandGrainOne) [reply src/Events/Extensions/CloudEventExtensions.cs]: The check should be done by the call point. Which I can't find in this PR. Where is the method being used?

Comment (acn-sbuad) [reply src/Events/Extensions/CloudEventExtensions.cs]: Currently only used by tests. Was wondering why I implemented it at this point myself. But we will need it for #338 as well

File: src/Events/Services/OutboundService.cs (hunk 2 above)
Comment (acn-sbuad) [reply src/Events/Services/OutboundService.cs:151]: Not a real possibility, We ensure all events have a resource before they are placed in inbound queue. I'll remove the check.

File: src/Events/Extensions/CloudEventExtensions.cs (hunk 1 above)
Comment (SandGrainOne) [reply src/Events/Extensions/CloudEventExtensions.cs]: I'm happy with `SetResourceIfNotDefined`.

File: test/Altinn.Platform.Events.Tests/TestingExtensions/CloudEventExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
+﻿using System;
+
+using Altinn.Platform.Events.Extensions;
+
+using CloudNative.CloudEvents;
+
+using Xunit;
+
+namespace Altinn.Platform.Events.Tests.TestingExtensions
+{
```
[the remark this answers is not in the document]
Comment (SandGrainOne) [reply test/Altinn.Platform.Events.Tests/TestingExtensions/CloudEventExtensionsTests.cs:10]: I like the new file scoped namespace declaration.

**Review by SandGrainOne — APPROVED** [review]

File: test/Altinn.Platform.Events.Tests/TestingExtensions/CloudEventExtensionsTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (acn-sbuad) [reply test/Altinn.Platform.Events.Tests/TestingExtensions/CloudEventExtensionsTests.cs]: For future reference, the commit suggestion must also remove a curly bracket at the end of the file to compile.

**Pull request merged**